### PR TITLE
deprecate: deprecate NLU JSON format

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- NLU data in JSON format is deprecated and will be removed in Rasa Open Source 4.0.
- fix ATO-26

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/migration-guide.mdx [hunk 1]
```diff
@@ -28,6 +28,13 @@ this new version for inference.
 
 Please check whether your trained model still performs as expected and retrain if needed.
 
+### NLU JSON Format
+
+[NLU training data](nlu-training-data.mdx) in JSON format is deprecated and will be
+removed in Rasa Open Source 4.0.
```
Comment (ancalita) [inline docs/docs/migration-guide.mdx:34]: Should we also add that the command will be removed too in 4.0?

File: rasa/shared/nlu/training_data/formats/rasa.py [hunk 2]
```diff
@@ -49,9 +94,20 @@ def read_from_json(self, js: Dict[Text, Any], **_: Any) -> "TrainingData":
...
 class RasaWriter(TrainingDataWriter):
+    """Dumps NLU data as Rasa JSON string."""
+
+    def __init__(self) -> None:
+        """Creates writer."""
+        super().__init__()
+        rasa.shared.utils.io.raise_deprecation_warning(
+            "NLU data in Rasa JSON format is deprecated and will be removed in Rasa "
+            "Open Source 4.0.0. Please convert your JSON NLU data to the "
+            "Rasa YAML format.",
```
Comment (ancalita) [inline rasa/shared/nlu/training_data/formats/rasa.py:105]: Should we also add here or whenever the `rasa data convert nlu` command is used that the command itself will be removed in 4.0?

**Review by ancalita — COMMENTED** [review]
Looks good 👍🏼 
Left some questions regarding the deprecation of the command itself.

File: docs/docs/migration-guide.mdx (hunk 1 above)
Comment (wochinge) [reply docs/docs/migration-guide.mdx:34]: We still support the convert command for users migration from LUIS/Watson/Dialogflow, don't we?

Comment (ancalita) [reply docs/docs/migration-guide.mdx:34]: I was confused as in the meeting I thought the consensus was towards removing the command, but your point makes sense.

**Pull request merged**